test/rendition.content.test.js

```diff
@@ -0,0 +1,189 @@
+        rendition.path = '\n\n';
+        result = await rendition.mimeType();
+        assert.strictEqual(result, null);
+    });
+
+    it('detects encoding of an existing an accessible file', async function () {
+        const instructions = { "fmt": "png", "target": "TargetName" };
+        const directory = "/";
+        let rendition = new Rendition(instructions, directory, 12);
+
+        // overwrite path to point to test files
+        rendition.path = './test/files/file.tif';
+        let result = await rendition.encoding();
+        assert.ok(result === null);
+        
+        rendition = new Rendition(instructions, directory, 12);
+        rendition.path = './test/files/file.txt';
+        result = await rendition.encoding();
+        assert.strictEqual(result, 'us-ascii');
+    });
+
+    it('gracefully handles not finding files when identifying encoding', async function () {
+        const instructions = { "fmt": "png", "target": "TargetName" };
+        const directory = "/";
+        let rendition = new Rendition(instructions, directory, 12);
+
+        // overwrite path to point to test files
+        rendition.path = './test/files/file-that-does-not-exist-and-should-therefore-not-be-here.bmp';
+        let result = await rendition.encoding();
+        assert.ok(result === null);
+
+        rendition = new Rendition(instructions, directory, 12);
+        rendition.path = '';
+        result = await rendition.encoding();
+        assert.ok(result === null);
+
+        rendition = new Rendition(instructions, directory, 12);
+        rendition.path = '  ';
+        result = await rendition.encoding();
+        assert.ok(result === null);
+
+        rendition = new Rendition(instructions, directory, 12);
+        rendition.path = '\n\n';
+        result = await rendition.encoding();
+        assert.ok(result === null);
+    });
+
+    it('verifies metadata works properly', async function () {
+        const instructions = { "fmt": "png", "target": "TargetName" };
+        const directory = "/";
+        const rendition = new Rendition(instructions, directory, 11);
+        rendition.path = './test/files/file.jpg';
+        const metadata = await rendition.metadata();
+
+        // test field by field
+        assert.strictEqual(metadata["repo:size"], 109472);
+        assert.strictEqual(metadata["repo:sha1"], "8b7ce94860836844eb17de009586fad2ca2fc8ad");
+        assert.strictEqual(metadata["tiff:imageWidth"], 512);
+        assert.strictEqual(metadata["tiff:imageHeight"], 288);
+        assert.strictEqual(metadata["dc:format"], "image/jpeg");
+        assert.ok(metadata["repo:encoding"] === undefined);
+    });
+
+    it('tests fallback from file command to file-type lib', async function () {
+        const failingFileCommand = async () => {
+            throw new Error('file command failure simulation');
+        };
+        rewiredRendition.__set__("getContentTypeInformation", failingFileCommand);
```

Comment:
I have nothing about doing white box testing when writing unit tests, even testing internal functions if some executions paths are hard to achieve. 
The thing I don't like about temporary overwriting functions/processes/libraries is that it is easy to forget they are overwritten in the test, and need to be restored. Especially when overwriting things broadly reusable, like `child_process`.
That is actually why I prefer rewire: when rewired, the rewired item under test is expected to not necessarily be in a good state. It's white box testing, and subject to change, sure. But if I would change the behavior, I probably also would want the test to fail so that I attend to the effects of the change.

Reply:
> temporary overwriting functions/processes/libraries is that it is easy to forget they are overwritten in the test
They should be setup and restored for each test, e.g. using `afterEach()`.